js/client-chat.js

```diff
@@ -185,10 +185,14 @@
 		// highlight
 
 		getHighlight: function (message) {
-			var highlights = Tools.prefs('highlights') || [];
-			if (!app.highlightRegExp) {
+			var highlights = Tools.prefs('highlights') || {};
+			if (highlights.constructor === 'Array') this.convertHighlights();
+			var thisroom = (this.type === 'battle' ? 'battle' : this.id);
+			if (!app.highlightRegExp || !app.highlightRegExp['global'] || !app.highlightRegExp[thisroom]) {
 				try {
-					this.updateHighlightRegExp(highlights);
+					if (!app.highlightRegExp) app.highlightRegExp = [];
+					if (!app.highlightRegExp['global']) this.updateHighlightRegExp(highlights['global'], 'global');
+					if (!app.highlightRegExp[thisroom]) this.updateHighlightRegExp(highlights[thisroom], thisroom);
```

Comment:
It seems like these three lines should probably be placed inside `updateHighlightRexExp`